scripts/g.extension/g.extension.py

```diff
@@ -2200,6 +2200,20 @@ def resolve_source_code(url=None, name=None, branch=None):
                 url_validated = True
             except:
                 pass
+            # Test if github repo exists (need to use API)
+            try:
+                open_url = urlopen('https://api.github.com/repos/{}/{}'.format(*url.split("/")[-2:]))
```

Comment:
Before checking if private repo url exists, it will be good checking if token has scope (permission) to get list of private repos (I'm going to add it in next commit). 

I'm using double quotes in python string interpolation in case `open_url = urlopen("https://api.github.com/repos/{}/{}".format(` and simple quotes in case `*url.split('/')`.